common/src/app/common/pages/changes_builder.cljc

```diff
@@ -323,7 +326,9 @@
                   :shapes (->> shapes reverse (mapv :id))}
 
            (some? index)
-           (assoc :index index))
+           (assoc :index index)
+           (:component-swap options)
+           (assoc :component-swap true))
```

Comment:
If you add a property to a change type, you must add it to the change spec, at `common/pages/changes' .